### PR TITLE
Add a build script for vector_add

### PR DESCRIPTION
**Pull request by abelsiqueira**
**Description**

Add a build script for the tests folder and the vector_add test/example.

**Related issues**:

- #116

**Instructions to review the pull request**

- Check that CHANGELOG.md has been updated if necessary

Clone and verify
```
cd $(mktemp -d --tmpdir cudawrappers-XXXXXX)
git clone https://github.com/nlesc-recruit/cudawrappers .
git checkout <this-branch>
cmake -S . -B build
make -C build
# prepend the following with `srun -N 1 -C TitanX --gres=gpu:1` on DAS
make -C build test
make -C build format # Nothing should change
make -C build lint # linting is broken until we fix the issues (see #92)
```

### CODE REVIEW

**Comment by abelsiqueira** [comment]
I have started a discussion on #116 which is blocking this.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# vector_add
+add_subdirectory(vector_add)
+
+# This command will be executed from the test folder, which is where the .cu file must be located
+add_custom_target(test
+  COMMAND vector_add/vector_add
+  DEPENDS vector_add
+)
```
Comment (fdiblen) [inline tests/CMakeLists.txt:8]:
```suggestion
)

```

File: README.dev.md [hunk 2]
```diff
@@ -190,7 +190,14 @@ where `<tool>` can be any of the following:
 
 ## Running the tests
```
Comment (fdiblen) [inline README.dev.md:191]:
```suggestion
## Running the tests on DAS
```

File: README.dev.md [hunk 3]
```diff
@@ -190,7 +190,14 @@ where `<tool>` can be any of the following:
...
 
-:construction:
+Enter the `build` directory and run `make test`.
+
+If you are running the tests on DAS, you can send create a job with command using `srun`.
```
Comment (fdiblen) [inline README.dev.md:195]:
```suggestion
If you are running the tests on DAS, you can run a job using `srun` command.
```

File: README.dev.md [hunk 4]
```diff
@@ -190,7 +190,14 @@ where `<tool>` can be any of the following:
...
+For instance,
+
+```shell
+srun -N 1 -C TitanX --gres=gpu:1 make test
+```
```
Comment (fdiblen) [inline README.dev.md:201]: This command will run the tests in one of the worker nodes with a GPU device.

Comment (fdiblen) [inline README.dev.md:201]:
```suggestion
This command will run the tests in one of the worker nodes with a GPU device.
```

**Review by fdiblen — CHANGES_REQUESTED** [review]
Looks great! I made a few suggestions.

**Review by fdiblen — APPROVED** [review]
Thanks @abelsiqueira :cookie:

**Comment by abelsiqueira** [comment]
Thanks for the review :broccoli:

**Pull request merged**